js/trek.js

```diff
@@ -0,0 +1,104 @@
+$(document).ready(function() {
+  $(document).foundation();
+
+  $('#trips, #single-trip, #reserve-trip, .error, #confirmation').hide();
+
+  const baseURL = 'https://trektravel.herokuapp.com/trips/';
+
+  /////// GET REQUEST FOR ALL TRIPS
+  const displayTrips = () => {
+    $.get(baseURL, successCallback)
+    .fail(function(){
+      $('.error').show();
+    });
+  };
+
+  /////// CALL BACK FOR ALL TRIPS
+  const successCallback = (response) => {
+    if ($("#table-body tr").length > 0) {
+      return;
+    }
+
+    $('#trip-name, #trip-country, #category, #weeks, #cost, #about').empty();
+    $('#single-trip, #reserve-trip, #confirmation').hide();
+    $('#trips').show();
+
+    for(let trip of response) {
+      if (trip.name === null || trip.name === " " || trip.continent === null || trip.weeks === null) {  continue; }
+      $('#table-body').append(
+        '<tr id=trip' + trip.id + '>' +
+          '<td>' + trip.id + '</td>' +
+          '<td>' + trip.name + '</td>' +
+          '<td>' + trip.continent + '</td>' +
+          '<td>' + trip.weeks + '</td>' +
+        '</tr>');
+      $('#trip' + trip.id).click((event) => {
+        displaySingleTrip(trip.id);
+      });
+    }
+  };
+
+  /// EVENT FOR ONE BUTTON
+  $('.main-button').click(displayTrips);
+
+  /////// CALL BACK FOR A SINGLE TRIP
+  const singleTripSuccessCallback = (response) => {
+    $('#table-body').empty();
+    $('#single-trip').show();
+    $('#trips, #confirmation').hide();
+
+    $('#trip-name').append(response.name.toUpperCase());
+    $('#trip-country').append(response.continent.toUpperCase());
+    $('#about').append(response.about);
+    $('#category').append('Category: ' + response.category.charAt(0).toUpperCase() + response.category.slice(1));
+    $('#weeks').append('Weeks: ' + response.weeks);
+    $('#cost').append('$' + response.cost);
+    $('.to-reserve').click(function(event) {
+      displayForm(response.id);
+    });
```

Comment:
Good use of a closure here to wrap / provide context to the call to `displayForm`.